module/extension/Director/src/director/run_task_pack.cpp

```diff
@@ -229,10 +229,50 @@ namespace module::extension {
         group.done = false;
         group.update_data();
 
+        // Check if a Wait command was emitted and schedule to run the Provider again
+        // Other tasks can run with Wait
+        // Wait should be removed and then readded at the end
+        TaskList non_wait_tasks;
+        TaskList wait_tasks;
+        for (const auto& t : requested_tasks) {
+            if (t->type == typeid(::extension::behaviour::Wait)) {
+                wait_tasks.push_back(t);
+
+                // Schedule the Provider to run again
+                // Get the time to wait for
+                auto wait_data                 = std::static_pointer_cast<::extension::behaviour::Wait>(t->data);
+                std::chrono::nanoseconds delay = std::chrono::nanoseconds(wait_data->time - NUClear::clock::now());
+
+                // If the delay is over, just run the provider
+                if (delay <= std::chrono::nanoseconds(0)) {
+                    run_task_on_provider(group.active_task, provider, RunReason::SUBTASK_DONE);
+                    continue;
+                }
+
+                // Otherwise, send it to the ChronoController to handle
+                // Make a weak pointer to the task so we can check if it still exists when the task is run
+                std::weak_ptr<component::DirectorTask> weak_task = t;
+                emit(std::make_unique<NUClear::dsl::operation::ChronoTask>(
+                    [this, provider, weak_task](const NUClear::clock::time_point&) {
+                        // Check if the task still exists
+                        if (weak_task.lock() != nullptr) {
+                            emit(std::make_unique<WaitFinished>(provider));
+                        }
+                        // Don't do anything else with this task
+                        return false;
+                    },
+                    NUClear::clock::now() + delay,
+                    -1));  // Our ID is -1 as we will remove ourselves
+            }
+            else {
+                non_wait_tasks.push_back(t);
+            }
+        }
+
         // Remove null data tasks from the list, this allows root tasks to be cleared
         TaskList tasks;
-        tasks.reserve(requested_tasks.size());
-        for (const auto& t : requested_tasks) {
+        tasks.reserve(non_wait_tasks.size());
```

Comment:
I would leave the names as requested tasks in case we add more categories in the future. Or maybe just filter out runnable tasks, but at the end just use the original vector

Reply:
I'm not sure I get what you mean
do you mean instead of `tasks`, make `requested_tasks` empty and use that?
or not have the `non_wait_tasks` and instead leave `requested_tasks` as-is, assuming Wait will disappear later?

Reply:
More the names of the variables, really the original requested tasks list should already have the wait tasks in it still... so you only need a filtered list of runnable tasks which has the wait tasks removed and is used for the rest. then at the end you just assign the original requested tasks to current tasks